thorn/src/kernel/mailbox.c

```diff
@@ -1,52 +1,56 @@
 #include "kernel/mailbox.h"
 
-bool mailbox_request (volatile unsigned int * data_ptr, channel_t channel) {
-    volatile unsigned int * buffer = data_ptr;
-
+bool mailbox_request (volatile unsigned int * buffer, channel_t channel) {
     if (DUMP_BUFFER) {
         LOG ("Dumping message buffer before sending:");
         int_dump ((unsigned int *) buffer);
         hex_dump ((byte_t *) buffer);
     }
 
-    // 28-bit address (MSB) and 4-bit value (LSB)
-    unsigned int outgoing = ((unsigned int) ((long) data_ptr) & ~0xF) | (channel & 0xF);
-
-    // Wait until we can write
-    while (get32 (MBOX0 + MBOX_STATUS) & MBOX_WRITE_FULL)
-        ;
+    mailbox_message_t message = {buffer, channel};
```

Comment:
```suggestion
    mailbox_message_t message = {buffer >> 4, channel};
```
see my explanation for `mailbox_message_to_register_value`

Reply:
This fails to compile with `invalid operands to binary >> (have ‘volatile unsigned int *’ and ‘int’)`. Tried to solve it with a cast to type long.

Reply:
preferably cast to `unsigned int` but both should work

Reply:
I updated the suggestion

Reply:
In that case I got a compiler warning because `unsigned int` is not necessarily same length as the pointer. This is why I took `long`.

Reply:
fair. In the end, we cast it to 32 bit implicitly